tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java

```diff
@@ -192,6 +193,16 @@ public synchronized ILiferayProject provide(Object adaptable) {
 				if (!LiferayWorkspaceUtil.isValidWorkspace(project) && LiferayNature.hasNature(project) &&
 					GradleProjectNature.isPresentOn(project)) {
 
+					LiferayCore liferayCore = LiferayCore.getDefault();
+
+					for (ILiferayProject liferayProject : liferayCore.getCachedProjects()) {
```

Comment:
@gamerson In the current logic, we will `provide` the `GradleWorkspaceProject`(for the case) several times because they have different `Cachekey`(IServer or IWorkspaceproject), and they share the same `IProject`. Once i make a `GradleWorkspaceProject` instance changed, however another won't. This is what i care, is this important?